kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -757,6 +756,31 @@ private void checkBigQuerySchemaUpdateConfigs() {
     }
   }
 
+  private void checkTimePartitioningConfigs(Map<String, String> properties) {
+    boolean decoratorSyntax = Optional.ofNullable((Boolean) ConfigDef.parseType(
+        BigQuerySinkTaskConfig.BIGQUERY_PARTITION_DECORATOR_CONFIG,
+        properties.get(BigQuerySinkTaskConfig.BIGQUERY_PARTITION_DECORATOR_CONFIG),
+        ConfigDef.Type.BOOLEAN
+    )).orElse(true);
+    boolean createTables = getBoolean(TABLE_CREATE_CONFIG);
+
+    if (!decoratorSyntax || !createTables) {
```

Comment:
> My understanding was that we will attempt to use decorator syntax to write to the table in `BigQuerySinkTaks::getRecordTable` even if create table was not specified.

This is correct; however, the time value specified for `timePartitioningType` only comes into play when automatically creating tables. If a table already exists, the connector will check for its partitioning type and adjust accordingly (in the case of this PR, by failing if decorator syntax is enabled and the time partitioning type isn't `DAY`).

I chose to fail only when table creation was enabled in order to avoid breaking existing connectors that might be mistakenly configured with a non-`DAY` partitioning type but without automatic table creation enabled. In that case the `timePartitioningType` property has no effect on the connector, so breaking because of how it's set seems unfair.

What are your thoughts?